scss/_patterns_code-snippet.scss

```diff
@@ -51,37 +52,69 @@ $color-snippet-heading-bg: rgba($color-x-dark, 0.08);
     }
 
     .p-code-snippet__header {
+      align-items: flex-start;
       background-color: $color-code-heading-background;
       display: flex;
       justify-content: space-between;
-      padding: $spv-inner--x-small--scaleable $sph-inner;
+      padding-left: $sph-inner;
+      padding-right: $sph-inner;
     }
 
     .p-code-snippet__title {
       @extend %vf-heading-5;
 
+      flex: 0 1 auto;
+      margin-bottom: $header-v-spacing;
+      overflow: hidden;
+      padding-top: $header-v-spacing;
+      text-overflow: ellipsis;
+      white-space: nowrap;
+    }
+
+    .p-code-snippet__dropdown {
+      background-color: $color-transparent;
+      border: 0;
+      color: $color-dark;
       margin-bottom: 0;
-      padding-top: 0;
+      margin-left: $sph-inner--small;
+      min-width: min-content;
+      padding-bottom: $header-v-spacing;
+      padding-left: $sph-inner;
+      padding-top: $header-v-spacing;
+      width: min-content;
     }
 
     .p-code-snippet__dropdowns {
+      display: flex;
       flex-shrink: 0;
-      margin-left: $sph-inner--small;
+
+      &:first-child {
+        // if no header is present, use this to
```

Comment:
Can't the dropdowns always have `margin-left: auto`?

Reply:
Good point